logstash-core/src/test/java/org/logstash/ackedqueue/QueueTest.java

```diff
@@ -1145,4 +1145,19 @@ public void firstUnackedPagePointToFullyAckedPurgedPage() throws Exception {
             assertFalse("Dangling page's checkpoint file should be removed", cp0.exists());
         }
     }
+
+    @Test
+    public void writeToClosedQueueException() throws Exception {
+        Settings settings = TestSettings.persistedQueueSettings(100, dataPath);
+        try {
+            Queue queue = new Queue(settings);
+            queue.open();
+            queue.write(new StringElement("First test string to be written in queue."));
+            queue.write(new StringElement("Second test string to be written in queue."));
+            queue.close();
+            queue.write(new StringElement("Third test string to be REJECTED to write in queue."));
+        } catch (QueueRuntimeException e) {
+            assertThat(e.getMessage(), CoreMatchers.containsString("Tried to write to a closed queue."));
+        }
```

Comment:
As-written, this test does not actually assert that an exception is thrown, only that _if_ it is thrown its message matches the desired pattern.

Additionally, any failure we encounter during the setup operations would be indistinguishable from a failure of the behaviour we are attempting to specify.

Normally, I like using JUnit's `Assert.assertThrows()` to assert for a specific block, which would look like this:

```suggestion
        Settings settings = TestSettings.persistedQueueSettings(100, dataPath);
        Queue queue = new Queue(settings);

        queue.open();
        queue.write(new StringElement("First test string to be written in queue."));
        queue.write(new StringElement("Second test string to be written in queue."));
        queue.close();

        final QueueRuntimeException qre = assertThrows(QueueRuntimeException.class, () -> {
            queue.write(new StringElement("Third test string to be REJECTED to write in queue."));
        });
        assertThat(qre.getMessage(), containsString("Tried to write to a closed queue."));
```

I don't think we have a recent enough JUnit activated, so a limited emulation of its behaviour is:

~~~ java

    @FunctionalInterface
    private interface Executable {
        void execute() throws Throwable;
    }

    static <T extends Throwable> T assertThrows(Class<T> expectedType, Executable executable) {
        try {
            executable.execute();
        } catch (Throwable actual) {
            Assert.assertThat(actual, Matchers.instanceOf(expectedType));
            return expectedType.cast(actual);
        }

        throw new AssertionError(String.format("Expected %s to be thrown, but nothing was thrown.", expectedType.getName()));
    }
~~~